### PR TITLE
Migrating ERC-20 Files to docs-app

### PR DESCRIPTION
**Pull request by Buddhika-Pathirathna**

### CODE REVIEW

**Comment by Buddhika-Pathirathna** [comment]
> The content isn't showing up on the local run, so I cannot review without this fix. Please check the sidebar.config.js, it needs some fixes and the label needs to change to **ERC-20** all-caps.

@GloreenS as we discussed yesterday, I checked this again cloning the branch to a separate folder, there also the content was visible for me. @ipopescu, could you please help us confirm this - Gloreen was unable to load the content and guess it could be some issue in sidebar.config.js,  I can't see it since it's working fine in my machine, can you please try to load it in your machine and check?

**Comment by ipopescu** [comment]
> ain cloning the branch to a separate folder, there also the content was visible for me. @ipopescu, could you please help us confirm this - Gloreen was unable to load the content and guess it could be some issue in sidebar.config.js, I can't see it since it's working fine in my machine, can you please try to load it in your machine and check?

@Buddhika-Pathirathna I am not able to load the docs-app locally. When I issue the command "yarn run start" I get the following error: 
```
- workflow/undelegate
    at Object.checkSidebarsDocIds (/Users/iulia/casper/docusaurus/docs-app/node_modules/@docusaurus/plugin-content-docs/lib/sidebars.js:339:19)
    at doLoadVersion (/Users/iulia/casper/docusaurus/docs-app/node_modules/@docusaurus/plugin-content-docs/lib/index.js:117:31)
    at async loadVersion (/Users/iulia/casper/docusaurus/docs-app/node_modules/@docusaurus/plugin-content-docs/lib/index.js:166:28)
    at async Promise.all (index 0)
    at async Object.loadContent (/Users/iulia/casper/docusaurus/docs-app/node_modules/@docusaurus/plugin-content-docs/lib/index.js:174:33)
    at async /Users/iulia/casper/docusaurus/docs-app/node_modules/@docusaurus/core/lib/server/plugins/index.js:58:46
    at async Promise.all (index 0)
    at async Object.loadPlugins (/Users/iulia/casper/docusaurus/docs-app/node_modules/@docusaurus/core/lib/server/plugins/index.js:57:27)
    at async Object.load (/Users/iulia/casper/docusaurus/docs-app/node_modules/@docusaurus/core/lib/server/index.js:186:82)
    at async start (/Users/iulia/casper/docusaurus/docs-app/node_modules/@docusaurus/core/lib/commands/start.js:44:19)
error Command failed with exit code 1.
```

File: source/docs/casper/workflow/erc-20/introduction.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+The four users are created in [Casper Integration](https://integration.cspr.live/) live site and following are the accounts created to demonstrate the use of ERC20 contract:
+
+-   [User A](https://integration.cspr.live/account/01f2dfc09a94ef7bce440f93a1bb6f17fdac0c913549927d452e7e91a376e9d20d)
+
+-   [User B](https://integration.cspr.live/account/015d4e20b5f7f687be80aed6e20960898b02c7549cc49ddf583224ecd894eca375)
+
+-   [User C](https://integration.cspr.live/account/0101fe69ae2012358e5ce8e8b39661d45d225251c4f19ebb7fc74b057637e65aa4)
+
+-   [User D](https://integration.cspr.live/account/0171bd7bac58780ce950007de575a472bcb30457e7b68427a6ed466568d71db1d6)
```
Comment (ipopescu) [inline source/docs/casper/workflow/erc-20/introduction.md:19]: Please remove the extra space between bullets.

File: source/docs/casper/workflow/erc-20/introduction.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Introduction
```
Comment (ipopescu) [inline source/docs/casper/workflow/erc-20/introduction.md:1]: Please use the content as it is here: https://github.com/casper-network/docs/tree/master/workflow/erc-20. I noticed that the content does not match. Further improvements or updates can be added as a separate PR.

File: source/docs/casper/workflow/erc-20/setup.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# Setup
+
+To initiate the set up:
```
Comment (ipopescu) [inline source/docs/casper/workflow/erc-20/setup.md:3]: Please use the same introduction found here: https://docs.casperlabs.io/en/latest/workflow/erc-20/setup.html

File: source/docs/casper/workflow/erc-20/transfers.md [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+
+## Transfer 50 ERC20 Tokens from A to B
+
+```
+    casper-client put-deploy -n http://3.143.158.19:7777
+    --chain-name integration-test
+    --secret-key ~/casper/ibm_demo/user_a/secret_key.pem
+    --session-hash hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180
+    --session-entry-point "transfer"
+    --session-arg "recipient:key='account-hash-9f81014b9c7406c531ebf0477132283f4eb59143d7903a2fae54358b26cea44b" --session-arg "amount:u256='50'"
```
Comment (ipopescu) [inline source/docs/casper/workflow/erc-20/transfers.md:11]: Move the second session-arg on a new line

File: source/docs/casper/workflow/erc-20/approvals.md [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+    --state-root-hash d6d4d3e59017dfc21e2c9a8e235e2a2b3a446284a066a1f1f6704559fbb35a66
+```
+
+<img src="/static/image/tutorials/erc-20/invokingAlToCheckCsAllowance.png" width="800"/>
+
+### Failure to Overspend C's Allowance\*\*
+
+```
+    casper-client put-deploy -n http://3.143.158.19:7777 \
+    --chain-name integration-test --secret-key ~/casper/ibm_demo/user_c/secret_key.pem \
```
Comment (ipopescu) [inline source/docs/casper/workflow/erc-20/approvals.md:238]: Move the secret key on a new line.

File: source/docs/casper/workflow/erc-20/transfers.md [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+
+We then call our utility contract to invoke the `balance_of` entry point and write the balance of User A to URef we sourced initially.
+
+```
+    casper-client put-deploy -n http://3.143.158.19:7777
+    --secret-key ~/casper/ibm_demo/user_a/secret_key.pem
+    --session-package-name "erc20_test_call"
+    --session-entry-point "check_balance_of"
+    --session-arg "token_contract:account_hash='account-hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180'"
+    --session-arg "address:key='account-hash-303c0f8208220fe9a4de40e1ada1d35fdd6c678877908f01fddb2a56502d67fd'" --chain-name integration-test
```
Comment (ipopescu) [inline source/docs/casper/workflow/erc-20/transfers.md:25]: Move chain-name on a new line.

File: source/docs/casper/workflow/erc-20/transfers.md [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+
+### Transfer of 20 ERC20 Tokens from B to C
+
+```
+    casper-client put-deploy -n http://3.143.158.19:7777
+    --chain-name integration-test
+    --secret-key ~/casper/ibm_demo/user_b/secret_key.pem
+    --session-hash hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180
+    --session-entry-point "transfer"
+    --session-arg "recipient:key='account-hash-89422a0f291a83496e644cf02d2e3f9d6cbc5f7c877b6ba9f4ddfab8a84c2670'" --session-arg "amount:u256='20'"
```
Comment (ipopescu) [inline source/docs/casper/workflow/erc-20/transfers.md:47]: Move the second session-arg on a new line.

File: source/docs/casper/workflow/erc-20/transfers.md [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+
+**Invoking `balance_of` for user A**
+
+```
+    casper-client put-deploy -n http://3.143.158.19:7777
+    --secret-key ~/casper/ibm_demo/user_a/secret_key.pem
+    --session-package-name "erc20_test_call"
+    --session-entry-point "check_balance_of"
+    --session-arg "token_contract:account_hash='account-hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180'"
+    --session-arg "address:key='account-hash-303c0f8208220fe9a4de40e1ada1d35fdd6c678877908f01fddb2a56502d67fd'" --chain-name integration-test
```
Comment (ipopescu) [inline source/docs/casper/workflow/erc-20/transfers.md:61]: Move chain-name on a new line.

File: source/docs/casper/workflow/erc-20/transfers.md [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+
+**Invoking `balance_of` for user B**
+
+```
+    casper-client put-deploy -n http://3.143.158.19:7777
+    --secret-key ~/casper/ibm_demo/user_a/secret_key.pem
+    --session-package-name "erc20_test_call"
+    --session-entry-point "check_balance_of"
+    --session-arg "token_contract:account_hash='account-hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180'"
+    --session-arg "address:key='account-hash-9f81014b9c7406c531ebf0477132283f4eb59143d7903a2fae54358b26cea44b'" --chain-name integration-test
```
Comment (ipopescu) [inline source/docs/casper/workflow/erc-20/transfers.md:81]: Move chain-name on a new line.

File: source/docs/casper/workflow/erc-20/transfers.md [hunk 10]
```diff
@@ -0,0 +1,251 @@
...
+
+**Invoking `balance_of` for user C**
+
+```
+    casper-client put-deploy -n http://3.143.158.19:7777
+    --secret-key ~/casper/ibm_demo/user_a/secret_key.pem
+    --session-package-name "erc20_test_call"
+    --session-entry-point "check_balance_of"
+    --session-arg "token_contract:account_hash='account-hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180'"
+    --session-arg "address:key='account-hash-89422a0f291a83496e644cf02d2e3f9d6cbc5f7c877b6ba9f4ddfab8a84c2670'" --chain-name integration-test
```
Comment (ipopescu) [inline source/docs/casper/workflow/erc-20/transfers.md:101]: Move chain-name on a new line.

File: source/docs/casper/workflow/erc-20/transfers.md [hunk 11]
```diff
@@ -0,0 +1,251 @@
...
+<img src="/static/image/tutorials/erc-20/invoke-bal-c.png" width="800"/>
+
+### Approve B to Spend 15 Tokens of A
+
+```
+    casper-client put-deploy -n http://3.143.158.19:7777
+    --chain-name integration-test
+    --secret-key ~/casper/ibm_demo/user_a/secret_key.pem
+    --session-hash hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180
+    --session-entry-point "approve" --session-arg "spender:key='account-hash-9f81014b9c7406c531ebf0477132283f4eb59143d7903a2fae54358b26cea44b'"
```
Comment (ipopescu) [inline source/docs/casper/workflow/erc-20/transfers.md:120]: Move session-arg on a new line.

File: source/docs/casper/workflow/erc-20/transfers.md [hunk 12]
```diff
@@ -0,0 +1,251 @@
...
+
+As we did for `balance_of` to read out the balance values, we must perform a similar invoking of the `allowance` entry point of the main ERC20 contract.
+
+```
+    casper-client put-deploy -n http://3.143.158.19:7777
+    --secret-key ~/casper/ibm_demo/user_a/secret_key.pem
+    --session-package-name "erc20_test_call"
+    --session-entry-point "check_allowance_of"
+    --session-arg "token_contract:account_hash='account-hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180'"
+    --session-arg "owner:key='account-hash-303c0f8208220fe9a4de40e1ada1d35fdd6c678877908f01fddb2a56502d67fd'" --session-arg "spender:key='account-hash-9f81014b9c7406c531ebf0477132283f4eb59143d7903a2fae54358b26cea44b'" --chain-name integration-test
```
Comment (ipopescu) [inline source/docs/casper/workflow/erc-20/transfers.md:135]: Move any argument with -- on a new line (here you have session-arg and chain-name).

File: source/docs/casper/workflow/erc-20/transfers.md [hunk 13]
```diff
@@ -0,0 +1,251 @@
...
+
+### Transfer 10 ERC20 Tokens from B’s Allowance to D
+
+```
+    casper-client put-deploy -n http://3.143.158.19:7777
+    --chain-name integration-test
+    --secret-key ~/casper/ibm_demo/user_b/secret_key.pem
+    --session-hash hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180
+    --session-entry-point "transfer_from"
+    --session-arg "owner:key='account-hash-303c0f8208220fe9a4de40e1ada1d35fdd6c678877908f01fddb2a56502d67fd'" --session-arg "recipient:key='account-hash-f32a2abc55316dc85a446a1c548674e03757974aaaf86e8b7d29947ae148eeca'" --session-arg "amount:u256='10'"
```
Comment (ipopescu) [inline source/docs/casper/workflow/erc-20/transfers.md:155]: Move the two additional session-arg to a new line.

File: source/docs/casper/workflow/erc-20/transfers.md [hunk 14]
```diff
@@ -0,0 +1,251 @@
...
+
+**Invoking `balance_of` for user A**
+
+```
+    casper-client put-deploy -n http://3.143.158.19:7777
+    --secret-key ~/casper/ibm_demo/user_a/secret_key.pem
+    --session-package-name "erc20_test_call"
+    --session-entry-point "check_balance_of"
+    --session-arg "token_contract:account_hash='account-hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180'"
+    --session-arg "address:key='account-hash-303c0f8208220fe9a4de40e1ada1d35fdd6c678877908f01fddb2a56502d67fd'" --chain-name integration-test
```
Comment (ipopescu) [inline source/docs/casper/workflow/erc-20/transfers.md:167]: Move chain-name to a new line.

File: source/docs/casper/workflow/erc-20/transfers.md [hunk 15]
```diff
@@ -0,0 +1,251 @@
...
+
+**Invoking `balance_of` for user B**
+
+```
+    casper-client put-deploy -n http://3.143.158.19:7777
+    --secret-key ~/casper/ibm_demo/user_a/secret_key.pem
+    --session-package-name "erc20_test_call"
+    --session-entry-point "check_balance_of"
+    --session-arg "token_contract:account_hash='account-hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180'"
+    --session-arg "address:key='account-hash-9f81014b9c7406c531ebf0477132283f4eb59143d7903a2fae54358b26cea44b'" --chain-name integration-test
```
Comment (ipopescu) [inline source/docs/casper/workflow/erc-20/transfers.md:187]: Move chain-name to a new line.

File: source/docs/casper/workflow/erc-20/transfers.md [hunk 16]
```diff
@@ -0,0 +1,251 @@
...
+
+**Invoking `balance_of` for user C**
+
+```
+    casper-client put-deploy -n http://3.143.158.19:7777
+    --secret-key ~/casper/ibm_demo/user_a/secret_key.pem
+    --session-package-name "erc20_test_call"
+    --session-entry-point "check_balance_of"
+    --session-arg "token_contract:account_hash='account-hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180'"
+    --session-arg "address:key='account-hash-89422a0f291a83496e644cf02d2e3f9d6cbc5f7c877b6ba9f4ddfab8a84c2670'" --chain-name integration-test
```
Comment (ipopescu) [inline source/docs/casper/workflow/erc-20/transfers.md:207]: Move chain-name to a new line.

File: source/docs/casper/workflow/erc-20/transfers.md [hunk 17]
```diff
@@ -0,0 +1,251 @@
...
+
+**Invoking `balance_of` for user D**
+
+```
+    casper-client put-deploy -n http://3.143.158.19:7777
+    --secret-key ~/casper/ibm_demo/user_a/secret_key.pem
+    --session-package-name "erc20_test_call"
+    --session-entry-point "check_balance_of"
+    --session-arg "token_contract:account_hash='account-hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180'"
+    --session-arg "address:key='account-hash-f32a2abc55316dc85a446a1c548674e03757974aaaf86e8b7d29947ae148eeca'" --chain-name integration-test
```
Comment (ipopescu) [inline source/docs/casper/workflow/erc-20/transfers.md:227]: Move chain-name to a new line.

File: source/docs/casper/workflow/erc-20/transfers.md [hunk 18]
```diff
@@ -0,0 +1,251 @@
...
+
+**Invoking `allowance` to check that it is 5**
+
+```
+    casper-client put-deploy -n http://3.143.158.19:7777
+    --secret-key ~/casper/ibm_demo/user_a/secret_key.pem
+    --session-package-name "erc20_test_call"
+    --session-entry-point "check_allowance_of"
+    --session-arg "token_contract:account_hash='account-hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180'"
+    --session-arg "owner:key='account-hash-303c0f8208220fe9a4de40e1ada1d35fdd6c678877908f01fddb2a56502d67fd'" --session-arg "spender:key='account-hash-9f81014b9c7406c531ebf0477132283f4eb59143d7903a2fae54358b26cea44b'" --chain-name integration-test
```
Comment (ipopescu) [inline source/docs/casper/workflow/erc-20/transfers.md:241]: Move session-arg and chain-name to a new line.

File: source/docs/casper/workflow/erc-20/approvals.md [hunk 19]
```diff
@@ -0,0 +1,271 @@
+# Approvals
+
+### Transfer 10 Tokens from an Allowance of only 5
+
+```
```
Comment (ipopescu) [inline source/docs/casper/workflow/erc-20/approvals.md:5]: For all code snippets, please add the highlighting. You can do this by adding the keyword "bash" after the ```.

Example:
```bash
    --session-arg "amount:u256='10'" \
    --payment-amount "10000000000"
```

File: config/sidebar.config.js [hunk 20]
```diff
@@ -2,14 +2,28 @@ module.exports = {
...
+            label: "ERC20 Sample Guide",
+            collapsible: true,
+            collapsed: true,
+            items: [
+                "workflow/erc20/introduction",
+                "workflow/erc20/setup",
+                "workflow/erc20/query",
+                "workflow/erc20/transfers",
+                "workflow/erc20/approvals",
+                "workflow/erc20/final-balance",
```
Comment (ipopescu) [inline config/sidebar.config.js:24]: @Buddhika-Pathirathna as you discovered, this part needs to be updated. I am adding this note here:

```
items: [
                "workflow/erc-20/introduction",
                "workflow/erc-20/setup",
                "workflow/erc-20/query",
                "workflow/erc-20/transfers",
                "workflow/erc-20/approvals",
                "workflow/erc-20/final-balance",
            ],
```

**Review by ipopescu — CHANGES_REQUESTED** [review]
@Buddhika-Pathirathna, thank you for your work on this section! 

- Please ensure that the content matches what we have today in the old docs repository, especially the _introduction_ and _setup_ sections. Maybe we have updated some of the introductions after you started the migration (I don't remember). 
- Please use _ERC-20_ instead of _ERC20_
- Update the _sidebar.config.js_ section
- Update the code blocks to have some highlighting, and let's also discuss with the team in the style guide
- Finally, some arguments in the code examples need to be brought on a new line, for visibility

**Comment by Buddhika-Pathirathna** [comment]
@ipopescu & @GloreenS Thank you so much for taking time for review, @ipopescu especially for checking each code block - I have addressed all the comments by you. @GloreenS - I have resolved the sidebar.config.js issue, so you must be able to load the content now.

**Review by ipopescu — APPROVED** [review]
@Buddhika-Pathirathna, I've noticed a couple of additional changes and I made them to speed up the merge:
- ~~Added a link to the guide on the main workflow page: https://casper.network/docs/workflow. See this: https://docs.casperlabs.io/en/latest/workflow/index.html~~ added as a mistake directly to main
- Removed the extra tabs in the code blocks. Let's discuss and hopefully keep the code blocks without extra space for easier reading and easier copy&paste.

File: source/docs/casper/workflow/erc-20-sample-guide/query.md [hunk 21]
```diff
@@ -0,0 +1,31 @@
+# Query the Contract Package
+
+Query the contract package hash which returns the package:
+
+```bash
+casper-client query-state -n http://3.143.158.19:7777
+--key hash-f9ae16d2a374e985c425c94c716c539070dc7d7907f119ba32d9684f4412a4ee
+--state-root-hash 4e8b0de303f834cb7c61bef148046e3de4446903bd15a395c9c37a6d96efe8c6
+```
```
Comment (GloreenS) [inline source/docs/casper/workflow/erc-20-sample-guide/query.md:9]: The escape character "\" is missing at the end of each line of the command (except the last one). Same for all commands in this file.
An example of how the line break should appear:
```bash
casper-client query-state -n http://3.143.158.19:7777 \
--key hash-f9ae16d2a374e985c425c94c716c539070dc7d7907f119ba32d9684f4412a4ee \
--state-root-hash 4e8b0de303f834cb7c61bef148046e3de4446903bd15a395c9c37a6d96efe8c6 
```

**Review by GloreenS — APPROVED** [review]
Looks good!
A minor change to the commands listed within the files. Also, ERC20 needs to be changed to ERC-20 in the **How To's** landing page. 
screenshot attached.
ERC-20_(10 11 2021_13 15 24_REC)

**Pull request merged**